### ISSUE TITLE
Fix `declaration-block-no-duplicate-properties` false negatives for interleaved non-consecutive duplicates with `ignore: ["consecutive-duplicates(-*)"]`

### ISSUE THREAD

**Issue by sarathfrancis90** [stylelint/stylelint#9325]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
  color: red;
  background: white;
  color: blue;
  background: black;
}
```

`color` and `background` are interleaved, and each appears as a **non-consecutive** duplicate.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-duplicate-properties": [true, { "ignore": ["consecutive-duplicates"] }]
  }
}
```

(The same happens with `ignore: ["consecutive-duplicates-with-different-syntaxes"]` and `["consecutive-duplicates-with-same-prefixless-values"]`.)

### How did you run Stylelint?

`npx stylelint test.css` and `npx stylelint --fix test.css`.

### Which Stylelint-related dependencies are you using?

stylelint 17.12.0 (also reproduces on the current `main`).

### What did you expect to happen?

With `ignore: ["consecutive-duplicates"]`, only *consecutive* duplicates are ignored; *non-consecutive* duplicates should still be reported. Both the second `color` and the second `background` are non-consecutive duplicates, so both should be flagged — and `--fix` should converge in a single pass.

### What actually happened?

Only the duplicate `color` is reported; the duplicate `background` is missed. As a result `--fix` is not idempotent:

- fix 1 → `a { background: white; color: blue; background: black; }`
- fix 2 → `a { color: blue; background: black; }`

The duplicate `background` is only removed on a second pass, which contradicts the rule's documentation (non-consecutive duplicates of the same property are listed as problems).

### Do you have a proposal to fix the bug?

Yes. Consecutiveness is inferred with the proxy `indexDuplicate === decls.length - 1`. When two properties are interleaved, the duplicated property is no longer the last "active" declaration, so this misclassifies a genuine non-consecutive duplicate and drops it. Tracking the previously-considered declaration index and comparing against it fixes detection and makes `--fix` single-pass. Happy to open a PR once this is confirmed.

**Comment by jeddy3** [comment]
Thanks for the report and for using the template.

I can replicate this with the demo.

Given:

```css
a { font-size: 16px; color: red; font-size: 1rem; color: blue; }
```

```js
export default {
  "rules": {
    "declaration-block-no-duplicate-properties": [true]
  }
};
```

Gives ([Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKBmEB2AXAtAM4CWAXjPCgIwBsADgB4DcKkANhAE6WcxQtY8RMhWq8Ati3ZdKAIzYBXGCwC+IADQgMxNjAByScXEQwGhuroB0YQoQ3gc2gOYIQpul1wpYGJArZewAA62CgoQSCc-jCEEZTBoWHhILBgbEicSLjEOPjyEGAA1vjYEPhQChbEYFkw+HScEHQwnNkxcSgA2rhRMAC6IWEqISpMIfaQ2M4AYlziWa6EuACeumzEeNaOxE6W4gBWdpqwdHaICWERS6sw63gdEem4MbgR6oPJ12sbBJPORLgkNgoBkoA8QE8XhERuMVEA)):

```
1:5-14 error Duplicate property "font-size"
1:22-27 error Duplicate property "color"
```

Where as:

```js
export default {
  "rules": {
    "declaration-block-no-duplicate-properties": [true, { "ignore": ["consecutive-duplicates"] }]
  }
};
```

Gives ([Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKBmEB2AXAtAM4CWAXjPCgIwBsADgB4DcKkANhAE6WcxQtY8RMhWq8Ati3ZdKAIzYBXGCwC+IADQgMxNjAByScXEQwGhuroB0YQoQ3gc2gOYIQpul1wpYGJArZewAA62CgoQSCc-jCEEZTBoWHhILBgbEicSLjEOPjyEGAA1vjYEPhQChbEYFkw+HScEHQwnNkxcSgA2rhRMOpoycROpbwdnRGQ2IQwYArZAG51FVU1uO0gALooKhshYSohKkwh9pPOAGJc4lmuhLgAnrpsxHjWjkOW4gBWdpqwdHZEAkwhE7o8YM88B0Iuk1ncIuo9skwU8XgQzkMiLgkNgoBkoNCQLCYrgIocTiogA):

```
1:5-14 error Duplicate property "font-size"
```

Where it should flag both properties, as neither is consecutive.
